src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Remove.cs

```diff
@@ -0,0 +1,579 @@
+            {
+                type = default;
+                return false;
+            }
+
+            return base.TryGetTypeForArgSuggestionAt(argIndex, out type);
+        }
+
+        public RemoveFunction()
+            : base("Remove", TexlStrings.AboutRemove, FunctionCategories.Behavior, DType.EmptyTable, 0, 2, int.MaxValue, DType.EmptyTable)
+        { 
+        }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
+        {
+            yield return new[] { TexlStrings.RemoveArg1, TexlStrings.RemoveArg2 };
+            yield return new[] { TexlStrings.RemoveArg1, TexlStrings.RemoveArg2, TexlStrings.RemoveArg2 };
+            yield return new[] { TexlStrings.RemoveArg1, TexlStrings.RemoveArg2, TexlStrings.RemoveArg2, TexlStrings.RemoveArg2 };
+        }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures(int arity)
+        {
+            if (arity > 2)
+            {
+                return GetGenericSignatures(arity, TexlStrings.RemoveArg1, TexlStrings.RemoveArg2);
+            }
+
+            return base.GetSignatures(arity);
+        }
+
+        public override IEnumerable<string> GetRequiredEnumNames()
+        {
+            return new List<string>() { LanguageConstants.RemoveFlagsEnumString };
+        }
+
+        public override bool IsLazyEvalParam(TexlNode node, int index, Features features)
+        {
+            // First argument to mutation functions is Lazy for datasources that are copy-on-write.
+            // If there are any side effects in the arguments, we want those to have taken place before we make the copy.
+            return index == 0;
+        }
+
+        public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DType[] argTypes, IErrorContainer errors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        {
+            Contracts.AssertValue(args);
+            Contracts.AssertAllValues(args);
+            Contracts.AssertValue(argTypes);
+            Contracts.Assert(args.Length == argTypes.Length);
+            Contracts.AssertValue(errors);
+            Contracts.Assert(MinArity <= args.Length && args.Length <= MaxArity);
+
+            bool fValid = base.CheckTypes(context, args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);
+            Contracts.Assert(returnType.IsTable);
+
+            DType collectionType = argTypes[0];
+            if (!collectionType.IsTable)
+            {
+                fValid = false;
+                errors.EnsureError(args[0], TexlStrings.ErrNeedCollection_Func, Name);
+            }
+
+            int argCount = argTypes.Length;
+            for (int i = 1; i < argCount; i++)
+            {
+                DType argType = argTypes[i];
+
+                if (!argType.IsRecord)
+                {
+                    if (argCount >= 3 && i == argCount - 1)
+                    {
+                        if (context.AnalysisMode)
```

Comment:
>context.AnalysisMode [](http://example.com/codeflow?start=28&length=20)

Why is this check for the stringly-typed enum done based on context.AnalysisMode? Shouldn't it be done based on Features.StronglyTypedBuiltinEnums?